dev/io.openliberty.jakartaee9.internal_fat/fat/src/io/openliberty/jakartaee9/internal/tests/EE9FeatureCompatibilityTest.java

```diff
@@ -0,0 +1,311 @@
+ *
+ * When a feature is expected to conflict with Jakarta EE 9 features, this test validates
+ * that the com.ibm.websphere.appserver.eeCompatible feature is one of the conflicts. This is
+ * important because there are messages that come up for eeCompatible conflicts to tell the
+ * customer that features are not compatible with Jakarta EE 9.
+ *
+ * As new value add features are added to Open Liberty this test will assume that they
+ * will support Jakarta EE 9. When Jakarta EE 10 is added there will need to be some additional
+ * logic to handle Jakarta EE 10 features as being expected to not run with EE 9 features.
+ */
+@SuppressWarnings("restriction")
+@RunWith(FATRunner.class)
+public class EE9FeatureCompatibilityTest extends FATServletClient {
+
+    private static final Class<?> c = EE9FeatureCompatibilityTest.class;
+
+    static final List<String> features = new ArrayList<>();
+
+    static final Set<String> nonEE9JavaEEFeatures = new HashSet<>();
+
+    static final Set<String> nonEE9MicroProfileFeatures = new HashSet<>();
+
+    static final Set<String> incompatibleValueAddFeatures = new HashSet<>();
+
+    static final String serverName = "jakartaee9.fat";
+    static final FeatureResolver resolver = new FeatureResolverImpl();
+    static FeatureRepository repository;
+
+    @Server("jakartaee9.fat")
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        File featureDir = new File(server.getInstallRoot() + "/lib/features/");
+        // If there was a problem building projects before this test runs, "lib/features" won't exist
+        if (featureDir != null && featureDir.exists()) {
+            for (File feature : featureDir.listFiles()) {
+                parseShortName(feature);
+            }
+        }
+
+        nonEE9JavaEEFeatures.addAll(EE7FeatureReplacementAction.EE7_FEATURE_SET);
+        nonEE9JavaEEFeatures.addAll(EE8FeatureReplacementAction.EE8_FEATURE_SET);
+
+        // appSecurity-1.0 is superceded by appSecurity-2.0 so it isn't in one of the replacement
+        // feature lists.  jaxb and jaxws 2.3 are EE related, but are noship features currently.
+        // jsp-2.2 is a EE6 feature that is included with open liberty.
+        nonEE9JavaEEFeatures.add("appSecurity-1.0");
+        nonEE9JavaEEFeatures.add("jaxb-2.3");
+        nonEE9JavaEEFeatures.add("jaxws-2.3");
+        nonEE9JavaEEFeatures.add("jsp-2.2");
+
+        // Remove test features that are in the FeatureReplacementActions
+        nonEE9JavaEEFeatures.remove("componenttest-1.0");
+        nonEE9JavaEEFeatures.remove("componenttest-2.0");
+        nonEE9JavaEEFeatures.remove("txtest-1.0");
+        nonEE9JavaEEFeatures.remove("txtest-2.0");
+        nonEE9JavaEEFeatures.remove("ejbTest-1.0");
+        nonEE9JavaEEFeatures.remove("ejbTest-2.0");
+
+        nonEE9MicroProfileFeatures.addAll(MicroProfileActions.MP10.getFeatures());
```

Comment:
the various MPXX FeatureSet objects would probably be better off as an enum that we can iterate over the values and the enum could indicate what EE versions it is associated with.  That way each time MP has a new release we don't have to update this code to add a new line.  Not really a blocker for this review.  Just an observation on where we could improve.

Reply:
MicroProfileActions.ALL is a `Set<FeatureSet>` that contains all versions